lib/http/headers.rb

```diff
@@ -49,16 +53,30 @@ def delete(name)
     # @param [Array<#to_s>, #to_s] value header value(s) to be appended
     # @return [void]
     def add(name, value)
```

Comment:
Would be nice to introduce optional keyword:

``` ruby
def add(name, value, normalize: nil)
```

Naturally, when normalize is false, no normalization should be done for the wire. When true - force normalization, and when nil - type dependent normalization.